stixcore/processing/decompression.py

```diff
@@ -0,0 +1,18 @@
+from stixcore.calibration.compression import decompress as algo_decompress
+
+
+def work_decompress(val, skm):
+    return algo_decompress(val, s=skm[0], k=skm[1], m=skm[2])
+
+
+def decompress(packet):
+    decompression_parameter = packet.get_decompression_parameter()
+    if not decompression_parameter:
+        return 0
+    c = 1
+    for param_name, (sn, kn, mn) in decompression_parameter.items():
+        skm = (sn if isinstance(sn, int) else packet.data.get(sn),
+               packet.data.get(kn),
+               packet.data.get(mn))
+        c += packet.data.work(param_name, work_decompress, skm)
```

Comment:
@samaloney you might wonder about that. your decompression method could deal with all all repeated NIX in the same packet at once. but then in generates a regular (n,m) array as output what might fail if there is no regular m.

so i process each array in the list as separate call within the work method. could also be helpfull for other processing steps later.